api/newsletter.ts

```diff
@@ -0,0 +1,49 @@
+import { NowRequest, NowResponse } from '@vercel/node'
+import fetch from 'node-fetch'
+
+const API_KEY = process.env.ECOMAIL_API_KEY || ''
+const API_URL = 'https://api2.ecomailapp.cz/lists/2/subscribe'
+
+export default async (request: NowRequest, response: NowResponse) => {
+  const body = request.body
+
+  if (!body || !body.email) {
+    response.status(404).send('Email is missing')
+    return
+  }
+
+  const data = JSON.stringify({
+    subscriber_data: {
+      email: body.email,
+      tags: ['web-subscribe-form'],
+    },
+  })
+
+  try {
+    const res = await fetch(API_URL, {
+      method: 'POST',
+      body: data,
+      headers: {
+        key: API_KEY,
+        'Content-Type': 'application/json',
+      },
+    })
+
+    // error from ecomail
+    if (!res.ok) {
+      const resData = await res.json()
+      response.status(500).json({ error: resData.message })
+      console.error('[HTTP ERROR]', resData.message)
+      return
+    }
+
+    // ok
+    response.json({
+      message: 'User subscription was successfull',
+    })
+  } catch (error) {
+    // network error
```

Comment:
console.log bychom neměli nechávat v produkčním kodu

Reply:
podle mě asi není třeba nechávat tento komentář. Z `catch` a `response.status(500)`, myslím, je vidět, že handlujeme error.

Reply:
Ten log jsem tam nechávala, protože jsem myslela, že případné errory dokážeme dohledat v logu Vercelu. Teď jsem se ale pořádně koukla do dokumentace a runtime logy se neuchovávají, tak jsem to smazala. Vercel uchovává pouze unhandled errors po dobu 30 dní, případně doporučují [Log Drain integraci](https://vercel.com/blog/refined-logging). Budeme to nějak řešit, nebo to mám nechat jak to je?

Reply:
Oke, za mě bych to takhle asi nechal bez `console.log` a pak udělal případně follow-up task na napojení logování těch serverless functions. Zatím jsem nepoužíval Vercel, takže nevím jak se tam řeší logovaní. Co myslíš @HormCodes

Reply:
Souhlas, za mne verze bez `console.log` je v pohodě (ESLint pak případně povoluje `console.error`). Pokud jde o logování ve Vercelu, tak nemám zkušenost, ale můžeme pak prozkoumat možnosti dodatečně.

Reply:
Plus mne ještě napadá, že ESLint by měl zahrnovat složku `api` a nevím jestli tomu tak je.